.github/workflows/generate-preview-link-eu.yml

```
@@ -0,0 +1,181 @@
+              run: echo '//npm.pkg.github.com/:_authToken=${{ secrets.READ_DERIV_COM_ORG_PACKAGES }}' >> .npmrc
+
+            - name: Get build output from master cache
+              uses: actions/cache/restore@e12d46a63a90f2fae62d114769bbf2a179198b5c
+              with:
+                  key: master-cache-public
+                  restore-keys: |
+                      master-cache-public-replica
+                  path: |
+                      sites/eu/.cache
+                      sites/eu/public
+
+            - name: Install Dependencies
+              uses: "./.github/actions/npm_install_from_cache"
+
+            - name: Build project
+              id: build-project
+              env:
+                  GATSBY_ENV: staging
+                  GATSBY_GROWTHBOOK_CLIENT_KEY: ${{ secrets.GATSBY_GROWTHBOOK_CLIENT_KEY }}
+                  GATSBY_GROWTHBOOK_DECRYPTION_KEY: ${{ secrets.GATSBY_GROWTHBOOK_DECRYPTION_KEY }}
+                  GATSBY_RUDDERSTACK_STAGING_KEY: ${{ secrets.GATSBY_RUDDERSTACK_STAGING_KEY }}
+                  GATSBY_RUDDERSTACK_PRODUCTION_KEY: ${{ secrets.GATSBY_RUDDERSTACK_PRODUCTION_KEY }}
+                  GATSBY_STRAPI_TOKEN: ${{ secrets.GATSBY_STRAPI_TOKEN }}
+                  GATSBY_GOOGLE_TAG_MANAGER_TRACKING_ID: ${{ secrets.GATSBY_GOOGLE_TAG_MANAGER_TRACKING_ID }}
+                  GATSBY_TRUSTPILOT_API_KEY: ${{ secrets.GATSBY_TRUSTPILOT_API_KEY }}
+                  GATSBY_HOTJAR_ID: ${{ secrets.GATSBY_HOTJAR_ID }}
+
+              run: npm run build:eu
+
+            - name: Retrieve PR information
+              env:
+                  EVENT_NUMBER: ${{ github.event.number }}
+                  EVENT_USERNAME: ${{ github.event.pull_request.user.login }}
+                  HEAD_REF: ${{ github.head_ref }}
+              run: |
+                  mkdir -p .pr
+                  echo "$EVENT_NUMBER" > .pr/NR
+                  echo "$EVENT_USERNAME" > .pr/USERNAME
+                  echo "$HEAD_REF" > .pr/BRANCHNAME
+
+            - name: Publish to Cloudflare Pages
+              id: publish-to-pages
+              env:
+                  CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_TEST_LINKS_API_TOKEN }}
+                  CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_TEST_LINKS_ACCOUNT_ID }}
+                  HEAD_BRANCH: ${{ github.head_ref }}
+              run: |
+                  echo "Installing Wrangler CLI"
+                  npm i -g wrangler
+                  echo "Deploying build to Cloudflare Pages"
+                  directory='sites/eu/public'
+                  projectName='deriv-com-preview-links'
+                  branch=$(echo "$HEAD_BRANCH" | head -c 20 | sed 's/[\/_\.]/-/g; s/[^a-zA-Z0-9]$/1/')
+                  cf_preview_url=$(wrangler pages deploy $directory --project-name=$projectName --branch=$branch > log.txt 2>&1; echo $?)
+                  echo "------"
+                  preview_url=https://eutestlink-$branch.deriv-com-preview-links.pages.dev
```

```suggestion
                  preview_url=https://eu--$branch.deriv-com-preview-links.pages.dev
```
the reason is that the url can get to long. so I just wanted to make it a bit shorter. IMO it is still clear.